keps/sig-apps/4212-declarative-node-maintenance/README.md

```diff
@@ -0,0 +1,2144 @@
+   to edit the PDB to account for the additional pod that needs to be spun to move the workload
+   from one node to another. This has been discussed in issue [kubernetes/kubernetes#66811](https://github.com/kubernetes/kubernetes/issues/66811)
+   and in issue [kubernetes/kubernetes#114877](https://github.com/kubernetes/kubernetes/issues/114877).
+2. Similar to the first point, it is difficult to use PDBs for applications that can have a variable
+   number of pods; for example applications with a configured horizontal pod autoscaler (HPA). These
+   applications cannot be disrupted during a low load when they have only pod. However, it is
+   possible to disrupt the pods during a high load without experiencing application downtime. If
+   the minimum number of pods is 1, PDBs cannot be used without blocking the node drain. This has
+   been discussed in issue [kubernetes/kubernetes#93476](https://github.com/kubernetes/kubernetes/issues/93476).
+3. Graceful deletion of DaemonSet pods is currently only supported as part of (Linux) graceful node
+   shutdown. The length of the shutdown is again not application specific and is set cluster-wide
+   (optionally by priority) by the cluster admin. This does not take into account 
+   `.spec.terminationGracePeriodSeconds` of each pod and may cause premature termination of
+   the application. This has been discussed in issue [kubernetes/kubernetes#75482](https://github.com/kubernetes/kubernetes/issues/75482)
+   and in issue [kubernetes-sigs/cluster-api#6158](https://github.com/kubernetes-sigs/cluster-api/issues/6158).
+4. There are cases during a node shutdown, when data corruption can occur due to premature node
+   shutdown. It would be great if applications could perform data migration and synchronization of 
+   cached writes to the underlying storage before the pod deletion occurs. This is not easy to
+   quantify even with pod's `.spec.shutdownGracePeriod`, as the time depends on the size of the data
+   and the speed of the storage. This has been discussed in issue [kubernetes/kubernetes#116618](https://github.com/kubernetes/kubernetes/issues/116618)
+   and in issue [kubernetes/kubernetes#115148](https://github.com/kubernetes/kubernetes/issues/115148).
+5. During the graceful node shutdown the kubelet terminates the pods in order of their priority.
+   The daemonset controller runs its own scheduling logic and creates the pods again. This causes a
+   race. Such pods should be removed and not recreated, but higher priority pods that have not yet
+   been terminated should be recreated if they are missing. This has been discussed in issue
+   [kubernetes/kubernetes#122912](https://github.com/kubernetes/kubernetes/issues/122912).
+6. The graceful node shutdown feature is not always reliable. If dbus or kubelet is restarted
+   during the shutdown, pods may be ungracefully terminated and lead to application disruption and
```

Comment:
Pods not only can be ungracefully terminated, but also sometimes the previously terminated workloads can be scheduled back on the node. kubelet will attempt to set the node status as such to allow scheduling on it to take place. That can also be undersired if not dangerous in some circumstances.

Reply:
ack, added